persistence/sql/migrations/20211019000001000000_merge_authentication_request_tables.sqlite.up.sql

```diff
@@ -0,0 +1,286 @@
+    acr                       TEXT         NULL,
+    login_authenticated_at    TIMESTAMP    NULL,
+    was_used                  INTEGER      NULL,
+    forced_subject_identifier VARCHAR(255) NULL DEFAULT '',
+    context                   TEXT         NULL DEFAULT '{}',
+    amr                       TEXT         NULL DEFAULT '',
+
+    consent_challenge_id      VARCHAR(40)  NULL,
+    consent_skip              INTEGER      NOT NULL,
+    consent_verifier          VARCHAR(40)  NOT NULL,
+    consent_csrf              VARCHAR(40)  NOT NULL,
+
+    ch_granted_scope         TEXT        NOT NULL,
+    ch_granted_at_audience   TEXT        NULL DEFAULT '',
+    ch_remember              INTEGER     NOT NULL,
+    ch_remember_for          INTEGER     NOT NULL,
+    ch_handled_at            TIMESTAMP   NULL,
+    ch_was_used              INTEGER     NOT NULL,
+    ch_error                 TEXT        NOT NULL,
+    ch_session_id_token      TEXT        NOT NULL,
+    ch_session_access_token  TEXT        NOT NULL,
+
+    CHECK (
+        state = 128 OR
+        state = 129 OR
+        state = 1 OR
+        (state = 2 AND (
+            remember IS NOT NULL AND
+            remember_for IS NOT NULL AND
+            error IS NOT NULL AND
+            acr IS NOT NULL AND
+            was_used IS NOT NULL AND
+            context IS NOT NULL AND
+            amr IS NOT NULL
+        )) OR
+        (state = 3 AND (
+            remember IS NOT NULL AND
+            remember_for IS NOT NULL AND
+            error IS NOT NULL AND
+            acr IS NOT NULL AND
+            was_used IS NOT NULL AND
+            context IS NOT NULL AND
+            amr IS NOT NULL
+        )) OR
+        (state = 4 AND (
+            remember IS NOT NULL AND
+            remember_for IS NOT NULL AND
+            error IS NOT NULL AND
+            acr IS NOT NULL AND
+            was_used IS NOT NULL AND
+            context IS NOT NULL AND
+            amr IS NOT NULL AND
+
+            consent_challenge_id IS NOT NULL AND
+            consent_verifier IS NOT NULL AND
+            consent_skip IS NOT NULL AND
+            consent_csrf IS NOT NULL
+        )) OR
+        (state = 5 AND (
+            remember IS NOT NULL AND
+            remember_for IS NOT NULL AND
+            error IS NOT NULL AND
+            acr IS NOT NULL AND
+            was_used IS NOT NULL AND
+            context IS NOT NULL AND
+            amr IS NOT NULL AND
+
+            consent_challenge_id IS NOT NULL AND
+            consent_verifier IS NOT NULL AND
+            consent_skip IS NOT NULL AND
+            consent_csrf IS NOT NULL AND
+
+            ch_granted_scope IS NOT NULL AND
+            ch_remember IS NOT NULL AND
+            ch_remember_for IS NOT NULL AND
+            ch_error IS NOT NULL AND
+            ch_session_access_token IS NOT NULL AND
+            ch_session_id_token IS NOT NULL AND
+            ch_was_used IS NOT NULL
+        ))
+    )
+);
+
+INSERT INTO hydra_flow (
```

Comment:
I would prefer to name this `hydra_oauth2_flow`, wdyt?

Reply:
> I will have to take a deeper dive into the persister logic, high level this looks very good! A major question for me is wether we really need to update the existing SDK tests? IMO everything should pass without modification to any existing tests (except for the persister tests of course)

There are two reasons why I made these changes:
1. The original tests used the same ID for the consent request and for the login request. I made the IDs different to ensure the new code doesn't mistake one for another, and also to make debugging the tests easier by ensuring it is apparent from the id whether it is for login or for consent. These changes were useful for development and are no longer necessary.
2. Some test changes are necessary because we now do slightly more validation, which means that some shortcuts used in tests are no longer possible. e.g. it is no longer possible to create a consent request without first creating a login request.

Should I go over the test changes and for each one either remove it or clarify why it is necessary?

> I would prefer to name this hydra_oauth2_flow, wdyt?

I don't have a preference, so let's make it hydra_oauth2_flow)

Reply:
> Should I go over the test changes and for each one either remove it or clarify why it is necessary?

I will take another look and mark the ones where I have questions - your explanation makes sense!